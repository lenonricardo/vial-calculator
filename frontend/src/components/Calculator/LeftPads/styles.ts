import styled from 'styled-components'

export const Pad = styled.div`
  width: var(--pad-width);
  height: var(--pad-height);
  color: var(--font-pads);
  display: flex;
  font-size: var(--default-font-size);
  align-items: center;
  justify-content: center;
  cursor: pointer;
  user-select: none;
`

export const PadWrapper = styled.div`
  display: flex;
  flex-direction: column;
`

export const BottomPads = styled.div`
  display: flex;
  flex-direction: row;
`

export const Pads = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(90px, 1fr));
  grid-template-rows: repeat(auto-fill, minmax(90px, 1fr));
  width: calc(450px - 90px - 90px);
  max-height: calc(450px - 90px);
`

export const TopPad = styled(Pad)`
  background-color: var(--secondary-operations);
`

export const DoublePad = styled(Pad)`
  width: calc(var(--pad-width) * 2);
`
