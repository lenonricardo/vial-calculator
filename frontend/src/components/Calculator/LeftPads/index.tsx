import { PadWrapper, TopPad, Pads, BottomPads, DoublePad, Pad } from './styles'

enum TopPadEnum {
  CE = 'CE',
  C = 'C',
  CHANGE_SIGN = '+/-'
}

enum NumberPadEnum {
  SEVEN = '7',
  EIGHT = '8',
  NINE = '9',
  FOUR = '4',
  FIVE = '5',
  SIX = '6',
  ONE = '1',
  TWO = '2',
  THREE = '3'
}

enum BottomPadEnum {
  ZERO = '0',
  DECIMAL_POINT = '.'
}

export default function LeftPads() {
  const topPads: TopPadEnum[] = Object.values(TopPadEnum)
  const numberPads: NumberPadEnum[] = Object.values(NumberPadEnum)

  return (
    <PadWrapper>
      <Pads>
        {topPads.map((label) => (
          <TopPad key={label}>{label}</TopPad>
        ))}

        {numberPads.map((label) => (
          <Pad key={label}>{label}</Pad>
        ))}
      </Pads>

      <BottomPads>
        <DoublePad>{BottomPadEnum.ZERO}</DoublePad>
        <Pad>{BottomPadEnum.DECIMAL_POINT}</Pad>
      </BottomPads>
    </PadWrapper>
  )
}
