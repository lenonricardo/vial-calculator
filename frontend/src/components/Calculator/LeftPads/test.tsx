import { render } from '@testing-library/react'

import LeftPads from '.'

describe('<LeftPads />', () => {
  it('should render the component', () => {
    const { container } = render(<LeftPads />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
