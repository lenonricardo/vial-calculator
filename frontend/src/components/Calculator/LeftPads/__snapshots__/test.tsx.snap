// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeftPads /> should render the component 1`] = `
.c2 {
  width: 90px;
  height: 90px;
  color: var(--font-pads);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 35px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(90px,1fr));
  grid-template-rows: repeat(auto-fill,minmax(90px,1fr));
  width: calc(450px - 90px - 90px);
  max-height: calc(450px - 90px);
}

.c3 {
  background-color: var(--secondary-operations);
}

.c5 {
  width: calc(90px + 90px);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    >
      CE
    </div>
    <div
      class="c2 c3"
    >
      C
    </div>
    <div
      class="c2 c3"
    >
      +/-
    </div>
    <div
      class="c2"
    >
      7
    </div>
    <div
      class="c2"
    >
      8
    </div>
    <div
      class="c2"
    >
      9
    </div>
    <div
      class="c2"
    >
      4
    </div>
    <div
      class="c2"
    >
      5
    </div>
    <div
      class="c2"
    >
      6
    </div>
    <div
      class="c2"
    >
      1
    </div>
    <div
      class="c2"
    >
      2
    </div>
    <div
      class="c2"
    >
      3
    </div>
  </div>
  <div
    class="c4"
  >
    <div
      class="c2 c5"
    >
      0
    </div>
    <div
      class="c2"
    >
      .
    </div>
  </div>
</div>
`;
